service/src/main/java/com/epam/esm/validator/GiftCertificateValidator.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.esm.validator;
+
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.entities.GiftCertificate;
+import com.epam.esm.entities.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+@Component
+public class GiftCertificateValidator {
```

Comment:
Suggestion: Lets refactor in next module to use https://spring.io/guides/gs/validating-form-input/

Reply:
Ok, I will refactor in next module